### PR TITLE
Env fixes & more

### PR DESCRIPTION
**Pull request by micrakow**

### CODE REVIEW

**Comment by micrakow** [comment]
@CAS-Linux-Jenkins retest this please

File: src/ocf_stats_builder.c [hunk 1]
```diff
@@ -221,6 +221,7 @@ static void _ocf_stats_part_fill(ocf_cache_t cache, ocf_part_id_t part_id,
 {
 	uint64_t cache_size, cache_line_size;
 
+
```
Comment (robertbaldyga) [inline src/ocf_stats_builder.c]: Unneeded empty line.

File: src/metadata/metadata.c [hunk 2]
```diff
@@ -39,9 +39,16 @@ int ocf_metadata_init(struct ocf_cache *cache,
 		return ret;
 	}
 
-	ocf_metadata_concurrency_init(&cache->metadata.lock);
+	ret = ocf_metadata_concurrency_init(&cache->metadata.lock);
+	if (ret) {
+		if (cache->metadata.iface.deinit) {
```
Comment (robertbaldyga) [inline src/metadata/metadata.c]: Unneeded brackets.

File: src/promotion/nhit/nhit_hash.c [hunk 3]
```diff
@@ -163,7 +163,10 @@ ocf_error_t nhit_hash_init(uint64_t hash_size, nhit_hash_t *ctx)
 		env_atomic_set(&new_ctx->ring_buffer[i].counter, 0);
 	}
 
-	env_spinlock_init(&new_ctx->rb_pointer_lock);
+	result = env_spinlock_init(&new_ctx->rb_pointer_lock);
+	if (result)
+		goto dealloc_locks;
```
Comment (arutk) [inline src/promotion/nhit/nhit_hash.c]: This won't work. i_locks is set to new_ctx->hash_entries, there will be out of bound access to array at line 177

File: src/ocf_freelist.c [hunk 4]
```diff
@@ -385,21 +385,29 @@ ocf_freelist_t ocf_freelist_init(struct ocf_cache *cache)
...
+
+spinlock_err:
+	while (i--)
+		env_spinlock_destroy(&freelist->lock[i]);
+free_allocs:
+	env_vfree(freelist->lock);
+	env_vfree(freelist->part);
+	env_vfree(freelist);
+	return NULL;
+
```
Comment (arutk) [inline src/ocf_freelist.c]: unnecessary empty line

File: src/metadata/metadata.c [hunk 5]
```diff
@@ -39,9 +39,16 @@ int ocf_metadata_init(struct ocf_cache *cache,
...
+			cache->metadata.iface.deinit(cache);
+		}
+		ocf_metadata_io_deinit(cache);
+		return ret;
+	}
 
-	return 0;
+	return ret;
```
Comment (arutk) [inline src/metadata/metadata.c]: i liked "return 0" better

File: src/concurrency/ocf_metadata_concurrency.c [hunk 6]
```diff
@@ -6,17 +6,34 @@
...
+		goto rwsem_err;
 
 	for (i = 0; i < OCF_IO_CLASS_MAX; i++) {
-		env_spinlock_init(&metadata_lock->partition[i]);
+		err = env_spinlock_init(&metadata_lock->partition[i]);
+		if (err)
+			goto spinlocks_err;
 	}
+
+	return err;
```
Comment (arutk) [inline src/concurrency/ocf_metadata_concurrency.c]: maybe "return 0;"?

**Pull request merged**